src/clib/pioc_support.c

```diff
@@ -3875,6 +3875,18 @@ int PIOc_openfile_retry(int iosysid, int *ncidp, int *iotype, const char *filena
         return check_mpi(NULL, file, mpierr, __FILE__, __LINE__);
     }
 
+    /* Broadcast IO type to all tasks. */
+    if ((mpierr = MPI_Bcast(&file->iotype, 1, MPI_INT, ios->ioroot, ios->my_comm)))
```

Comment:
Is this required if the user did not request a retry (do we need an if(retry){} here?)?

Reply:
Also, this change is probably required only because we now care about the switched iotype on all procs (the prev commit). If iotype is only required on the I/O procs this change is probably not needed.
(if that is the case this should be mentioned in the commit message)

Reply:
> Is this required if the user did not request a retry (do we need an if(retry){} here?)?

Yes, we can check retry flag here. If retry flag is false, the file type will never be modified.

Reply:
> Also, this change is probably required only because we now care about the switched iotype on all procs (the prev commit). If iotype is only required on the I/O procs this change is probably not needed. (if that is the case this should be mentioned in the commit message)

OK, I will add this info in the commit message.